rts/lra/lra-filters/src/main/java/io/narayana/lra/filter/ServerLRAFilter.java

```diff
@@ -394,26 +394,36 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
 
         try {
             if (current != null && isJaxRsCancel(requestContext, responseContext)) {
-                lraClient.cancelLRA(current);
-                if (current.toASCIIString().equals(
-                        Current.getLast(requestContext.getHeaders().get(LRA_HTTP_HEADER)))) {
-                    // the callers context was ended so invalidate it
-                    requestContext.getHeaders().remove(LRA_HTTP_HEADER);
-                }
+                try {
+                    lraClient.cancelLRA(current);
+                } catch (NotFoundException ignore) {
+                    // must already be cancelled
+                } finally {
+                    if (current.toASCIIString().equals(
+                            Current.getLast(requestContext.getHeaders().get(LRA_HTTP_HEADER)))) {
+                        // the callers context was ended so invalidate it
+                        requestContext.getHeaders().remove(LRA_HTTP_HEADER);
+                    }
 
-                if (toClose != null && toClose.toASCIIString().equals(current.toASCIIString())) {
-                    toClose = null; // don't attempt to finish the LRA twice
+                    if (toClose != null && toClose.toASCIIString().equals(current.toASCIIString())) {
+                        toClose = null; // don't attempt to finish the LRA twice
+                    }
                 }
             }
 
             if (toClose != null) {
-                lraClient.closeLRA(toClose);
-                requestContext.getHeaders().remove(LRA_HTTP_HEADER);
-
-                if (toClose.toASCIIString().equals(
-                        Current.getLast(requestContext.getHeaders().get(LRA_HTTP_HEADER)))) {
-                    // the callers context was ended so invalidate it
+                try {
+                    lraClient.closeLRA(toClose);
+                } catch (NotFoundException ignore) {
+                    // must already be closed
```

Comment:
similarly as above, it would be good to have a trace log here (with information what LRA id has to be already closed)

Reply:
This is the same comment and I have the same response.

Reply:
Same comment as above^^